docs/design/architecture.md

```diff
@@ -0,0 +1,18 @@
+# Overview
```

Comment:
I think we should either drop this doc or defer it to another PR that updates the content as it is woefully out of date. It isn't entirely wrong but it's missing a lot of relevant detail and emphasizes parts of the architecture that are minor while not touching on important aspects.

Reply:
Removed the commit that added architecture.md file.